### PR TITLE
[Shadow] Add Mind Sear and Voidbolt to guide and make the guide the default view

### PR DESCRIPTION
**Pull request by DoxAshe**
### Description

The insanity graph is now working in the guide view.
This fixes Mind Sear's target hits and adds analysis for Mind Sear in the guide view.
It adds cast efficiency to Void Bolt when Void Eruption is taken.
Shadow's Death and Madness talent now correctly counts and resets Shadow Word: Death in spellusable.  Its suggestions were from the Shadowlands version of the talent, so it is now removed. 
Major and short cooldown cast efficiencies now are colored based on their suggestion thresholds in the guide view.

### Motivation

With these additions, the guide view finally has all the information the old version has, plus addition analysis and better presentation.  Having the guide view be the default view will be better.

### Screenshot(s):

Insanity Graph

MSGuide

ColorCoded Efficiency

### CODE REVIEW

File: src/analysis/retail/priest/shadow/modules/talents/MindSear.tsx [hunk 1]
```diff
@@ -28,17 +45,70 @@ class MindSear extends Analyzer {
...
+    // Sometimes there can be a slight difference in timestamp even for the same ticks. This gives a plenty of tolerance while never risking catching the next set.
+    if (timeDiff > 100) {
+      this.ticks += 1;
+      this.recentTicks += 1;
+    }
+
+    this.time = this.checkTime;
+  }
+
+  _tallyLastCast() {
```
Comment (ToppleTheNun) [inline src/analysis/retail/priest/shadow/modules/talents/MindSear.tsx]:
```suggestion
  private _tallyLastCast() {
```

File: src/analysis/retail/priest/shadow/modules/talents/MindSear.tsx [hunk 2]
```diff
@@ -28,17 +45,70 @@ class MindSear extends Analyzer {
...
+  onFightEnd() {
+    this._tallyLastCast(); // make sure the last cast is closed out
+  }
+
+  onCast(event: CastEvent) {
+    this._tallyLastCast(); // make sure the previous cast is closed out
+    this._trackNewCast(event);
+  }
+
+  _trackNewCast(event: CastEvent) {
```
Comment (ToppleTheNun) [inline src/analysis/retail/priest/shadow/modules/talents/MindSear.tsx]:
```suggestion
  private _trackNewCast(event: CastEvent) {
```

**Review by ToppleTheNun — CHANGES_REQUESTED** [review]
Couple of very minor changes, otherwise looks good!

File: src/analysis/retail/priest/shadow/modules/talents/MindSear.tsx (hunk 1 above)
Comment (DoxAshe) [reply src/analysis/retail/priest/shadow/modules/talents/MindSear.tsx]: I've added private to _tallyLastCast

File: src/analysis/retail/priest/shadow/modules/talents/MindSear.tsx (hunk 2 above)
Comment (DoxAshe) [reply src/analysis/retail/priest/shadow/modules/talents/MindSear.tsx]: I've added private to _trackNewCast

**Review by ToppleTheNun — APPROVED** [review]

**Pull request merged**